server/controllers/manage/lostBedsController.ts

```diff
@@ -32,9 +32,13 @@ export default class LostBedsController {
 
       const { startDate } = DateFormats.dateAndTimeInputsToIsoString(req.body, 'startDate')
       const { endDate } = DateFormats.dateAndTimeInputsToIsoString(req.body, 'endDate')
-      const numberOfBeds = Number(req.body.numberOfBeds)
 
-      const lostBed: NewLostBed = { ...req.body.lostBed, startDate, endDate, numberOfBeds }
+      const lostBed: NewLostBed = {
```

Comment:
If I'm understanding you right we'd have to know what nunjucks is sending us wouldn't we? Would be very useful but I don't think it's impossible. 
I haven't tried them but I think it could be better if we used (TypedRequestHandlers)[https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/33cd75ca0e0ac9c2ad7681eeb37beb3acf6cff91/server/@types/express/index.d.ts#L22]

Reply:
@pezholio I think [zod](https://zod.dev/?id=basic-usage) may do what we want? Think it's in use elsewhere in HMPPS. 
You can go 'all in' with it but I wonder if we could just use it in our controllers to check the data we are getting from the views is correct

Reply:
Oh sweet! Yeah, that sounds like a plan 👍